pex/resolver.py

```diff
@@ -333,6 +333,7 @@ def resolve(self, resolvables, resolvable_set=None):
         requirement = resolvable.requirement
       else:
         requirement = distribution.as_requirement()
+        requirement.extras = tuple(resolvable.extras())
```

Comment:
> 1. What do you anticipate the type to be in the `else` block? Because you're calling `resolvable.extras()`,

The `extras` method is defined on the base class; so the name resolvables is in fact typey here - its a collection of resolvables and we know nothing (nor need know anything) more:
https://github.com/pantsbuild/pex/blob/87ea9b5d8d37de92debb5090690c80a623515718/pex/resolvable.py#L123-L125

> 2. It's confusing to me the difference between `ResolvableRequirement` vs `pex.third_party.pkg_resources.Requirement`. I think the idea of those 4 lines is that we want requirement to always have type `pex.third_party.pkg_resources.Requirement`, right?

Right. How about I surface types in the next line down?:
https://github.com/pantsbuild/pex/blob/87ea9b5d8d37de92debb5090690c80a623515718/pex/resolver.py#L336-L337 No matter the branch `ResolvedDistribution` needs to be constructible with requirement and distribution; so documenting / checking the types there seems like a good central way to tighten this up.

Reply:
> Right. How about I surface types in the next line down?

That would help a lot. Thank you!

As an aside, I would love to add type comments to Pex even more than to Pants. I've found one of my biggest time sucks with these Pex reviews is figuring out what things are supposed to be, e.g. what `requirement` is (possibly because I have less experience with this codebase). I know type comments are especially ugly—and I don't have the bandwidth to do this fully—but the surface area is much smaller than Pants so maybe feasible to do. Especially valuable imo if it would help to get more Pex contributors.

Reply:
Well, that is also partly a weird side-effect of doing more reviews than commits (49/12). Doing commits its fairly easy to get type info as things stand, esp. if using something like pycharm. That said, I definitely wouldn't block efforts towards this.